### PR TITLE
Don't use input traits in neo

### PR DESCRIPTION
**Pull request by robertbastian**
We have `NeoGetField` for this. 

Also adds `format_timezone`.

### CODE REVIEW

File: components/datetime/src/format/datetime.rs [hunk 1]
```diff
@@ -67,7 +67,7 @@ use writeable::{Part, Writeable};
 /// ```
 #[derive(Debug, Copy, Clone)]
 pub struct FormattedDateTime<'l> {
```
Comment (sffc) [inline components/datetime/src/format/datetime.rs:69]: Note: `FormattedDateTime` is being deleted in #5543. But, it looks like this PR is mostly independent of that.

File: components/datetime/src/format/neo.rs [hunk 2]
```diff
@@ -2217,13 +2213,21 @@ where
     ///     "The date is: November 20, 1700 Anno Domini"
     /// );
     /// ```
-    pub fn format_date<T>(&self, datetime: &'a T) -> FormattedDateTimePattern<'a>
+    pub fn format_date<I>(&self, datetime: &'a I) -> FormattedDateTimePattern<'a>
```
Comment (sffc) [inline components/datetime/src/format/neo.rs:2216]: This function is obsolete and should be deleted. Well, it should be replaced with a format function that binds on all of the input markers of `R`, like in NeoFormatter.

**Review by sffc — DISMISSED** [review]

File: components/datetime/src/format/neo.rs (hunk 2 above)
Comment (robertbastian) [reply components/datetime/src/format/neo.rs:2216]: How is this obsolete but `format_time` is not?

File: components/datetime/src/format/datetime.rs (hunk 1 above)
Comment (robertbastian) [reply components/datetime/src/format/datetime.rs:69]: I had this on top of #5543 but I don't want to block on it because it doesn't seem ready.

Comment (sffc) [reply components/datetime/src/format/datetime.rs:69]: I wanted to finish it on Friday but ran out of time... now it will be bitrot. Oh well :)

**Review by sffc — APPROVED** [review]
Not the final solution, but an incrementally positive change

File: components/datetime/src/format/neo.rs (hunk 2 above)
Comment (robertbastian) [reply components/datetime/src/format/neo.rs:2216]: I see a use case for these, you have a type that implements time fields, but you only want to format the date.

Comment (sffc) [reply components/datetime/src/format/neo.rs:2216]: The fields that get formatted are determined by the pattern you feed into this type, though.

**Pull request merged**